tools/mksurfdata_esmf/src/mkharvestMod.F90

```diff
@@ -253,8 +248,7 @@ subroutine mkharvest(file_mesh_i, file_data_i, mesh_o, pioid_o, constant, ntime,
        end if
     end do
 
-    ! If constant model, clean up the mapping
-    if (constant) then
+    if (.not. present(ntime)) then  ! ...else we will reuse it
```

Comment:
@ekluzek suggests:
Could pass both ntime and end_time so as to also destroy routehandle in last year.
May not be worth the trouble, but it's a good practice.

Reply:
Looking into this:
mksurfdata.F90 doesn't have explicit knowledge of end_time and instead exits the time loop like this:
```
if (root_task) then
           read(nfdyn, '(A195,1x,I4)', iostat=ier) string, year
           if (ier /= 0) end_of_fdynloop = .true.
        end if
        call mpi_bcast(end_of_fdynloop, 1, MPI_LOGICAL, 0, mpicom, ier)
        if (end_of_fdynloop) then
           EXIT
        end if
```
So I think this needs a more extensive refactor to accomplish the suggestion of the previous post.

Reply:
I agree, let's not worry about the clean at the end.